python/source/io.rst

```diff
@@ -0,0 +1,423 @@
+========================
+Reading and Writing Data
+========================
+
+Recipes related to reading and writing data from disk using
+Apache Arrow.
+
+.. contents::
+
+Write a Parquet file
+====================
+
+.. testsetup::
+
+    import numpy as np
+    import pyarrow as pa
+
+    arr = pa.array(np.arange(100))
+
+Given an array with all numbers from 0 to 100
+
+.. testcode::
+
+    print(f"{arr[0]} .. {arr[-1]}")
+
+.. testoutput::
+
+    0 .. 99
+
+To write it to a Parquet file, as Parquet is a columnar format,
+we must create a :class:`pyarrow.Table` out of it,
```

Comment:
Parquet being columnar is not quite the reason, it's that Parquet is a file format for storing tables with named columns, so we have to put the array in a Table or RecordBatch, giving that array a name within the object's schema

Reply:
I see how using "columnar" in this context might be misleading, I used it to mean "contains multiple columns" but I see that one might understand it as "column oriented/major" which is not what I was referring to here. I'll rephrase this as `"as Parquet is a format that contains multiple named columns"` which should be clear and explicitly point out they are named as you suggested.